lib/runtime-core/src/trampoline_x64.rs

```diff
@@ -246,44 +248,49 @@ impl TrampolineBufferBuilder {
         &mut self,
         target: unsafe extern "C" fn(*const CallContext, *const u64) -> u64,
         context: *const CallContext,
-        num_params: u32,
+        params: &[Type],
```

Comment:
Any reason this shouldn't just take the full function signature? As it happens it only used to need the number of parameters, now it needs the whole list of the types, but isn't that an implementation detail?

Reply:
This function does not prepend the WebAssembly-specific `vm::Ctx` argument and is just a general-purpose trampoline generator, therefore should not take a WebAssembly `FuncSig`. I added a `returns` argument just in case of future compatibility.